### PR TITLE
Added a way to meassure performance under different conditions

### PR DESCRIPTION
**Pull request by dgomezTT**
### Ticket
[Link to Github Issue](https://github.com/tenstorrent/tt-metal/issues/22039)

### Problem description
Right now the metrics we provide for bert are not really clear or final. We don't have cached scenarios or metrics for the whole run.

### What's changed
Added iterations over the models to present how the model behaves under different circumstances such as:
no cache
cold cache
warm cache

Each model implementation will show more specific data

### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable)
- [ ] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [ ] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml) CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: models/demos/bert/demo/demo.py [hunk 1]
```diff
@@ -83,85 +86,98 @@ def run_bert_question_and_answering_inference(
...
+            max_length=sequence_size,
+            padding="max_length",
+            truncation=True,
+            return_attention_mask=True,
+            return_token_type_ids=True,
+            return_tensors="pt",
+        )
+
+        position_ids = positional_ids(config, bert_input.input_ids)
+        profiler.start(f"preprocessing_input")
```
Comment (jmalone-tt) [inline models/demos/bert/demo/demo.py:119]: I think "end_to_end_inference" should start here instead. For apples-to-apples comparison, we assume users have already put inputs through a tokenizer

File: models/demos/bert/demo/demo.py [hunk 2]
```diff
@@ -83,85 +86,98 @@ def run_bert_question_and_answering_inference(
...
+            *ttnn_bert_inputs,
+            parameters=parameters,
+        )
+        profiler.end(f"inference_time")
+        profiler.start(f"post_processing_output_to_string")
+
+        tt_output = (
+            ttnn.to_torch(ttnn.from_device(tt_output)).reshape(batch_size, 1, sequence_size, -1).to(torch.float32)
+        )
+
```
Comment (jmalone-tt) [inline models/demos/bert/demo/demo.py:141]: Similar to the above, I think it is reasonable to have "end_to_end_inference" end here, before we extract start and end logits

**Review by jmalone-tt — COMMENTED** [review]
Looks good to me other than small adjustments to end_to_end_inference

File: models/demos/bert/demo/demo.py (hunk 2 above)
Comment (dgomezTT) [reply models/demos/bert/demo/demo.py:141]: both of them addressed. Thanks for clarifying it!

**Review by ayerofieiev-tt — APPROVED** [review]

**Review by jmalone-tt — APPROVED** [review]

**Review by TT-BrianLiu — APPROVED** [review]
